### ISSUE TITLE
Unexpected JSX interface resolution with namespaces

### ISSUE THREAD

**Issue by wnayes** [Microsoft/TypeScript#27570]
**TypeScript Version:**  3.2.0-dev.201xxxxx

When using a [custom JSX factory](https://www.typescriptlang.org/docs/handbook/jsx.html#factory-functions), the `JSX` interface lookup behaves unexpectedly when the factory is on a namespace.

For example, if I define my factory function as follows:
```ts
@jsx MyCompany.Name.Space.JSXRenderer.createElement
```

**Actual behavior:**

It currently looks for the `JSX` interface in the following locations:

- `MyCompany.JSX`
- global `JSX`

**Expected behavior:**

I would expect it to look in the following locations:

- `MyCompany.Name.Space.JSXRenderer.JSX`
- global `JSX`

Basically, it's more likely that the `JSX` interface would be defined at the same level as `createElement`.

This appears to be due to the use of `getFirstIdentifier` in `checker.ts` `getJsxNamespace`.

**Comment by RyanCavanaugh** [comment]
Not sure what to say; the documentation is pretty clear and we can't match the first-guess expectations of all users 🙃
